### ISSUE TITLE
"add_field" and "tags" configuration options have no effect

### ISSUE THREAD

**Issue by passing** [logstash-plugins/logstash-input-beats#12]
I am using this configuration with logstash 2.0.0 ...
```
input {
  beats {
    port => 5044
    add_field => { "application" => "beats" }
    tags => [ "beats" ]
  }
}

output {
  stdout {
    codec  => rubydebug {
      metadata => true
    }
  }
}
```
... and send some data using 'topbeat'

In the output I see neither the field 'application' nor the tag 'beats'. This would be helpful to create filters for this input when other inputs are used in parallel

**Comment by ph** [comment]
I will check this, thanks for reporting it.

**Comment by urso** [comment]
@passing you can add tags to the shipper configuration in topbeat.

**Comment by ph** [comment]
@urso I think this is also a problem with how we decorate the events, I haven't tested it yet.

**Comment by robin13** [comment]
I confrim the `add_fields` but only when adding to @metadata:

```
input {
    beats {
        port => 5044
        add_field => {
            "[@metadata][inputtype]" => "beats"
           "foo" => bar
        }
        tags => [ "baz" ]
    }
```

results in the `foo` field being added, but not the `[@metadata][inputtype]`.
The `tags` option works as expected for me.

**Comment by ph** [comment]
@robin13 I believe you were using filebeat for your tests.

I believe that when we are using `topbeat` as the source it doesn't go through the `codec` part.

https://github.com/logstash-plugins/logstash-input-beats/blob/master/lib/logstash/inputs/beats.rb#L134

**Comment by robin13** [comment]
@ph True - I was using filebeat.

**Comment by ph** [comment]
Fixed in 0.9.4
